relayer/contracts/generate.go

```diff
@@ -1,6 +1,5 @@
 //go:generate bash -c "jq .abi ../../core/packages/contracts/out/OpaqueProof.sol/OpaqueProof.json | abigen --abi - --type OpaqueProof --pkg contracts --out opaque_proof.go"
 //go:generate bash -c "jq .abi ../../core/packages/contracts/out/BeefyClient.sol/BeefyClient.json | abigen --abi - --type BeefyClient --pkg contracts --out beefy_client.go"
-//go:generate bash -c "jq .abi ../../core/packages/contracts/out/InboundQueue.sol/InboundQueue.json | abigen --abi - --type InboundQueue --pkg contracts --out inbound_queue.go"
-//go:generate bash -c "jq .abi ../../core/packages/contracts/out/OutboundQueue.sol/OutboundQueue.json | abigen --abi - --type OutboundQueue --pkg contracts --out outbound_queue.go"
+//go:generate bash -c "jq .abi ../../core/packages/contracts/out/Gateway.sol/Gateway.json | abigen --abi - --type Gateway --pkg contracts --out gateway.go"
```

Comment:
Actually, we should be generating bindings for `contracts/src/interfaces/IGateway.sol`.

This is how `GatewayProxy.sol`, `Gateway.sol`, and `IGateway.sol` all relate to each other:

* `GatewayProxy.sol`: The contract at the "front of the house" that users interact with. This is the proxy contract.
* `Gateway.sol`: `GatewayProxy.sol` forwards messages to this implementation or "logic" contract.
* `IGateway.sol`: The API exposed by the proxy & implementation contract.

When configuring the relayer, make sure to use the address of `GatewayProxy.sol` as the gateway's address.

Reply:
Right, will use the interface. We're using the proxy address [here](https://github.com/Snowfork/snowbridge/pull/888/commits/3bca8e8981373833b6b2dfa2fa1ee17b572c8b7b)

Reply:
I've kept the name `Gateway` for the Go bindings, even though they're generated from the IGateway interface. Seems sensible as the interface is the only thing we'll use from the relayers.